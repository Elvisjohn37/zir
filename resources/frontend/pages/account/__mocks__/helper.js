let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
