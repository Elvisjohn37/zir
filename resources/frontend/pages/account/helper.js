export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
