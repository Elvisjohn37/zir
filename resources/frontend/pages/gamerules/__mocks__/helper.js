let mockedModule = jest.createMockFromModule('../helper');

mockedModule.extractProductGameType = (gameRulesList) => {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!extractedItems[indexMap[item.productName]].gameTypes.includes(item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
};

module.exports = mockedModule;
