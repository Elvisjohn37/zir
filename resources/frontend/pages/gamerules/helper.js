import lodash from 'lodash';

function extractProductGameType(gameRulesList) {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!lodash.includes(extractedItems[indexMap[item.productName]].gameTypes, item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
}

export { extractProductGameType };
