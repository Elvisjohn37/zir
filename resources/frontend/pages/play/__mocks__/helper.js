let postMessageHandler = () => jest.fn();

let setPostMessageHandler = () => jest.fn();

let resetPostMessageHandler = () => jest.fn();

resetPostMessageHandler();

// _________________________________________

let windowResponseData,
	isMobile = false;

let windowResponse = () => windowResponseData;

let resetWindowResponse = () => {
	windowResponseData = {
		result: true,
		data: {
			url: 'https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641',
			title: 'Beat The Bobbies',
			maxPayout: '700000000.000000',
			isMobile: false,
			viewport: 'desktop',
			productID: 'zy',
			balanceInfo: { playableBalance: '1457.800000', currencyCode: 'ZWD' },
		},
	};
};

let setWindowResponse = (data) => {
	windowResponseData = {
		...windowResponseData,
		data: {
			...windowResponseData.data,
			...data,
		},
	};
};

let setErrorResponse = () => {
	windowResponseData = {
		...windowResponseData,
		result: false,
	};
};

resetWindowResponse();

// _________________________________________

module.exports = { postMessageHandler, windowResponse, setWindowResponse, resetWindowResponse, setErrorResponse };
