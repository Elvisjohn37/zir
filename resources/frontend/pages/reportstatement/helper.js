import { getStatementReport as getStatementReportAjax } from 'frontend/ajax/backend';

export function getStatementReport(
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) {
	setStatementDataLoading(true);

	getStatementReportAjax(
		statementRange,
		!isMounted,
		(response) => {
			setStatementDataLoading(false);
			setStatementData({
				content: response.data.data.content,
				footer: response.data.data.footer,
			});
			if (!isMounted) {
				setDateRange({
					isLoading: false,
					range: response.data.data.dateRange,
				});
			}
		},
		() => {
			setStatementDataError(true);
			setStatementDataLoading(false);
		}
	);
}
