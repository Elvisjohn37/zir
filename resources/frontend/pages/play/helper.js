import { windowResponse } from 'frontend/utils/helper';

let postMessageHandlerWrapper = () => {
	let allowedOrigin = windowResponse() ? windowResponse().data.url : '';

	return function ({ data, origin }) {
		let originUrl = new URL(origin);
		let gameUrl = new URL(allowedOrigin);

		if ((gameUrl.origin == originUrl.origin || window.location.host == originUrl.host) && data == 'closeGame') {
			window.location.href = '/';
		}
	};
};

let postMessageHandler = postMessageHandlerWrapper();

export { postMessageHandler, windowResponse };
