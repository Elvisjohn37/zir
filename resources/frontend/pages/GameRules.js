import React from 'react';
import Link from '@mui/material/Link';
import { NavLink } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import { LangDataProvider } from 'frontend/components/Language';
import styles from './GameRules.module';
import { GameRulesConsumer } from 'frontend/contexts/GameRules';
import GameRulesLoader from 'frontend/components/GameRulesLoader';
import lodash from 'lodash';
import { extractProductGameType } from './gamerules/helper';
import GameRulesErrorMessage from 'frontend/components/GameRulesErrorMessage';

export default function GameRules() {
	return (
		<div className={styles.gameRules}>
			<GameRulesConsumer>
				{({ gameRulesState }) =>
					gameRulesState.isLoading ? (
						<GameRulesLoader />
					) : gameRulesState.isError ? (
						<GameRulesErrorMessage />
					) : (
						<LangDataProvider category="gamerules" className="testGamingRules">
							<div className={styles.gamingRules}>
								<h3>
									<FormattedMessage id="applicability.title" />
								</h3>
								<div className={styles.content}>
									<FormattedMessage id="applicability.description1" />
									<Link href="/" className="testHomePage">
										<FormattedMessage id="applicability.description2" />
									</Link>
									<FormattedMessage id="applicability.description3" />
								</div>
								<h3>
									<FormattedMessage id="generalRules" />
								</h3>
								<ol>
									<h3>
										<li>
											<FormattedMessage id="liability.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="liability.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="playForFun.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="playForFun.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="personalUse.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="personalUse.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="malfunctions.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="malfunctions.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="smartPlayer.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="smartPlayer.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="complaintsRng.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="complaintsRng.description1" />
										<Link href="mailto:support@sbobet.com">
											<FormattedMessage id="complaintsRng.description2" />
										</Link>
										<FormattedMessage id="complaintsRng.description3" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="complaintsLg.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="complaintsLg.description1" />
										<br />
										<br />
										<FormattedMessage id="complaintsLg.description2" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="maxWin.title" />
										</li>
									</h3>
									<div className={styles.content}>
										<FormattedMessage id="maxWin.description" />
									</div>
									<h3>
										<li>
											<FormattedMessage id="liveCasinoGames.title" />
										</li>
									</h3>

									<div className={styles.content}>
										<FormattedMessage id="liveCasinoGames.description1" />
										<br />
										<br />
										<FormattedMessage id="liveCasinoGames.description2" />
										<br />
										<br />

										<ol>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#BACCARAT'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_baccarat" />
												</Link>
											</li>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#ROULETTE'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_roulette" />
												</Link>
											</li>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#SICBO'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_sicbo" />
												</Link>
											</li>
											<li>
												<Link
													href={gameRulesState.config.liveDealerGameLink + '#BLACKJACK'}
													target="_blank"
													className={styles.liveCasinoLink}
												>
													<FormattedMessage id="casinoGames.live_dealer_blackjack" />
												</Link>
											</li>
											{extractProductGameType(gameRulesState.gameRulesList).map((item, index) => (
												<li key={index} className={styles.gameRulesProduct}>
													<FormattedMessage
														id={`casinoGames.${lodash.camelCase(item.productName)}`}
														defaultMessage={item.productName.toUpperCase()}
													/>
													<ol>
														{item.gameTypes.map((gameTypeItem, gameTypeIndex) => (
															<li key={`gameType${gameTypeIndex}`} className={styles.gameRulesGameType}>
																<Link
																	component={NavLink}
																	to={`/gamerules/${lodash.kebabCase(item.productName)}/${lodash.kebabCase(
																		gameTypeItem
																	)}`}
																>
																	<FormattedMessage
																		id={`casinoGames.${lodash.camelCase(gameTypeItem)}`}
																		defaultMessage={gameTypeItem}
																	/>
																</Link>
															</li>
														))}
													</ol>
												</li>
											))}
										</ol>
									</div>
								</ol>
							</div>
						</LangDataProvider>
					)
				}
			</GameRulesConsumer>
		</div>
	);
}
