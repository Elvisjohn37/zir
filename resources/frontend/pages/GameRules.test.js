import React from 'react';
import { render } from '@testing-library/react';
import GameRules from './GameRules';
import { setGameRulesState, resetGameRulesState, getGameRulesState } from 'frontend/contexts/GameRules';
import { extractProductGameType } from './gamerules/helper';
import * as lodash from 'lodash';

jest.mock('@mui/material/Link');
jest.mock('react-router-dom');
jest.mock('react-intl');
jest.mock('frontend/components/Language');
jest.mock('frontend/contexts/GameRules');
jest.mock('frontend/components/GameRulesLoader');
jest.mock('lodash');
jest.mock('./gamerules/helper');
jest.mock('frontend/components/GameRulesErrorMessage');
jest.mock('classnames');

describe('GameRules Component', () => {
	beforeEach(() => {
		resetGameRulesState();
	});

	it('should match snapshot', () => {
		let { asFragment } = render(<GameRules />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should display loading indicator while game rules list is fetching', () => {
		setGameRulesState({
			isLoading: true,
		});
		let { getByTestId, queryByTestId, rerender } = render(<GameRules />);

		expect(getByTestId('GameRulesLoader')).toBeInTheDocument;

		resetGameRulesState();
		rerender(<GameRules />);

		expect(queryByTestId('GameRulesLoader')).not.toBeInTheDocument();
	});

	it('should redirect live casino game rules to correct URL', () => {
		let { baseElement } = render(<GameRules />);
		let liveCasinoLink = baseElement.querySelectorAll('.liveCasinoLink');
		let gameRulesState = getGameRulesState();
		liveCasinoLink.forEach((link) => {
			expect(link.getAttribute('href').startsWith(gameRulesState.config.liveDealerGameLink)).toBeTruthy();
		});
	});

	it('should display zircon product and game type link to group of game rules', () => {
		let { baseElement } = render(<GameRules />);
		let products = baseElement.querySelectorAll('.gameRulesProduct');
		let gameRulesList = extractProductGameType(getGameRulesState().gameRulesList);

		gameRulesList.map((gameRule, index) => {
			let product = products[index];
			let productTitle = product.querySelector('[data-testid="FormattedMessage"]');
			let allGameTypesLink = product.querySelectorAll('a');

			expect(productTitle).toHaveAttribute('id', `casinoGames.${lodash.camelCase(gameRule.productName)}`);

			gameRule.gameTypes.map((gameType, gameTypeIndex) => {
				let gameTypeLink = allGameTypesLink[gameTypeIndex];
				let gameTypeTitle = gameTypeLink.querySelector('[data-testid="FormattedMessage"]');

				expect(gameTypeLink).toHaveAttribute(
					'to',
					`/gamerules/${lodash.kebabCase(gameRule.productName)}/${lodash.kebabCase(gameType)}`
				);
				expect(gameTypeTitle).toHaveAttribute('id', `casinoGames.${lodash.camelCase(gameType)}`);
			});
		});
	});

	it('should display error message when gamerules response is not valid', () => {
		setGameRulesState({
			isError: true,
		});
		let { getByTestId } = render(<GameRules />);

		expect(getByTestId('GameRulesErrorMessage')).toBeInTheDocument();
	});
});
