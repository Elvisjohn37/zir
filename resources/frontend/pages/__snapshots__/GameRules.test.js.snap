// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameRules Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="gamerules"
    class="testGamingRules"
    data-testid="LangDataProvider"
  >
    <div>
      gamerules
    </div>
    <div
      class="gamingRules"
    >
      <h3>
        <div
          data-testid="FormattedMessage"
          id="applicability.title"
        >
          <div>
            applicability.title
          </div>
        </div>
      </h3>
      <div
        class="content"
      >
        <div
          data-testid="FormattedMessage"
          id="applicability.description1"
        >
          <div>
            applicability.description1
          </div>
        </div>
        <a
          class="testHomePage"
          data-testid="Link"
          href="/"
        >
          <div
            data-testid="FormattedMessage"
            id="applicability.description2"
          >
            <div>
              applicability.description2
            </div>
          </div>
        </a>
        <div
          data-testid="FormattedMessage"
          id="applicability.description3"
        >
          <div>
            applicability.description3
          </div>
        </div>
      </div>
      <h3>
        <div
          data-testid="FormattedMessage"
          id="generalRules"
        >
          <div>
            generalRules
          </div>
        </div>
      </h3>
      <ol>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="liability.title"
            >
              <div>
                liability.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="liability.description"
          >
            <div>
              liability.description
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="playForFun.title"
            >
              <div>
                playForFun.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="playForFun.description"
          >
            <div>
              playForFun.description
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="personalUse.title"
            >
              <div>
                personalUse.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="personalUse.description"
          >
            <div>
              personalUse.description
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="malfunctions.title"
            >
              <div>
                malfunctions.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="malfunctions.description"
          >
            <div>
              malfunctions.description
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="smartPlayer.title"
            >
              <div>
                smartPlayer.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="smartPlayer.description"
          >
            <div>
              smartPlayer.description
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="complaintsRng.title"
            >
              <div>
                complaintsRng.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="complaintsRng.description1"
          >
            <div>
              complaintsRng.description1
            </div>
          </div>
          <a
            data-testid="Link"
            href="mailto:support@sbobet.com"
          >
            <div
              data-testid="FormattedMessage"
              id="complaintsRng.description2"
            >
              <div>
                complaintsRng.description2
              </div>
            </div>
          </a>
          <div
            data-testid="FormattedMessage"
            id="complaintsRng.description3"
          >
            <div>
              complaintsRng.description3
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="complaintsLg.title"
            >
              <div>
                complaintsLg.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="complaintsLg.description1"
          >
            <div>
              complaintsLg.description1
            </div>
          </div>
          <br />
          <br />
          <div
            data-testid="FormattedMessage"
            id="complaintsLg.description2"
          >
            <div>
              complaintsLg.description2
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="maxWin.title"
            >
              <div>
                maxWin.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="maxWin.description"
          >
            <div>
              maxWin.description
            </div>
          </div>
        </div>
        <h3>
          <li>
            <div
              data-testid="FormattedMessage"
              id="liveCasinoGames.title"
            >
              <div>
                liveCasinoGames.title
              </div>
            </div>
          </li>
        </h3>
        <div
          class="content"
        >
          <div
            data-testid="FormattedMessage"
            id="liveCasinoGames.description1"
          >
            <div>
              liveCasinoGames.description1
            </div>
          </div>
          <br />
          <br />
          <div
            data-testid="FormattedMessage"
            id="liveCasinoGames.description2"
          >
            <div>
              liveCasinoGames.description2
            </div>
          </div>
          <br />
          <br />
          <ol>
            <li>
              <a
                class="liveCasinoLink"
                data-testid="Link"
                href="http://info.sbobet.com/article/AA-00363/41/#BACCARAT"
                target="_blank"
              >
                <div
                  data-testid="FormattedMessage"
                  id="casinoGames.live_dealer_baccarat"
                >
                  <div>
                    casinoGames.live_dealer_baccarat
                  </div>
                </div>
              </a>
            </li>
            <li>
              <a
                class="liveCasinoLink"
                data-testid="Link"
                href="http://info.sbobet.com/article/AA-00363/41/#ROULETTE"
                target="_blank"
              >
                <div
                  data-testid="FormattedMessage"
                  id="casinoGames.live_dealer_roulette"
                >
                  <div>
                    casinoGames.live_dealer_roulette
                  </div>
                </div>
              </a>
            </li>
            <li>
              <a
                class="liveCasinoLink"
                data-testid="Link"
                href="http://info.sbobet.com/article/AA-00363/41/#SICBO"
                target="_blank"
              >
                <div
                  data-testid="FormattedMessage"
                  id="casinoGames.live_dealer_sicbo"
                >
                  <div>
                    casinoGames.live_dealer_sicbo
                  </div>
                </div>
              </a>
            </li>
            <li>
              <a
                class="liveCasinoLink"
                data-testid="Link"
                href="http://info.sbobet.com/article/AA-00363/41/#BLACKJACK"
                target="_blank"
              >
                <div
                  data-testid="FormattedMessage"
                  id="casinoGames.live_dealer_blackjack"
                >
                  <div>
                    casinoGames.live_dealer_blackjack
                  </div>
                </div>
              </a>
            </li>
            <li
              class="gameRulesProduct"
            >
              <div
                data-testid="FormattedMessage"
                defaultmessage="GAMES"
                id="casinoGames.games"
              >
                <div>
                  casinoGames.games
                </div>
              </div>
              <ol>
                <li
                  class="gameRulesGameType"
                >
                  <a
                    data-testid="NavLink"
                    to="/gamerules/games/table-games"
                  >
                    <div
                      data-testid="FormattedMessage"
                      defaultmessage="Table Games"
                      id="casinoGames.tableGames"
                    >
                      <div>
                        casinoGames.tableGames
                      </div>
                    </div>
                  </a>
                </li>
                <li
                  class="gameRulesGameType"
                >
                  <a
                    data-testid="NavLink"
                    to="/gamerules/games/card-games"
                  >
                    <div
                      data-testid="FormattedMessage"
                      defaultmessage="Card Games"
                      id="casinoGames.cardGames"
                    >
                      <div>
                        casinoGames.cardGames
                      </div>
                    </div>
                  </a>
                </li>
              </ol>
            </li>
            <li
              class="gameRulesProduct"
            >
              <div
                data-testid="FormattedMessage"
                defaultmessage="MULTIPLAYER"
                id="casinoGames.multiplayer"
              >
                <div>
                  casinoGames.multiplayer
                </div>
              </div>
              <ol>
                <li
                  class="gameRulesGameType"
                >
                  <a
                    data-testid="NavLink"
                    to="/gamerules/multiplayer/card-games"
                  >
                    <div
                      data-testid="FormattedMessage"
                      defaultmessage="Card Games"
                      id="casinoGames.cardGames"
                    >
                      <div>
                        casinoGames.cardGames
                      </div>
                    </div>
                  </a>
                </li>
              </ol>
            </li>
          </ol>
        </div>
      </ol>
    </div>
  </div>
</DocumentFragment>
`;
